### PR TITLE
Add `--acyclic` option

### PR DESCRIPTION
**Pull request by smoelius**
@regexident Are you still open to adding #18?

I tried to implement it in a way that was faithful to https://github.com/regexident/cargo-modules/issues/18#issuecomment-391614170.

For example, here is the output from a run on [cargo](https://crates.io/crates/cargo):
```
$ cargo modules generate graph --acyclic --layout=none --package cargo --lib
Error: Circular dependency between `cargo::util::config` and `cargo::util::toml`.

┌> cargo::util::config
│  └─> cargo::util::auth
│      └─> cargo::util
│          └─> cargo::util::toml
└──────────────┘
```

Nits are welcome, though.

### CODE REVIEW

**Comment by smoelius** [comment]
Re the rustfmt failure: I included a rustfmt.toml file to make it easier to diff tri_color.rs against https://github.com/rust-lang/rust/blob/925dc37313853f15dc21e42dc869b024fe488ef3/compiler/rustc_data_structures/src/graph/iterate/mod.rs.

Is that silly? Should we just live with an ugly diff?

I'm not sure what to make of "test with minimal versions" failure.

**Comment by smoelius** [comment]
Also, I added an "MIT OR Apache-2.0" license header to tri_color.rs. But based on my understanding of those licenses, that file could be licensed MPL.

**Comment by smoelius** [comment]
> I'm not sure what to make of "test with minimal versions" failure.

Ah. It's failing because of `bitvec` dependencies:
- https://github.com/ferrilab/bitvec/blob/5fb855073acc2ed045094ed89d8daf8c765f0135/Cargo.toml#L61
- https://github.com/ferrilab/bitvec/blob/5fb855073acc2ed045094ed89d8daf8c765f0135/Cargo.toml#L72

**Comment by smoelius** [comment]
I'm going to hold back on trying to push further fixes. Please let me know how you would like to proceed.

File: src/graph/tri_color.rs [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+//! The original rustc implementation appears to be due to @ecstatic-morse, though all errors
+//! resulting from porting or modification are mine (@smoelius).
+//!
+//! [can-a-3-color-dfs-be-used-to-identify-cycles-not-just-detect-them]: https://cs.stackexchange.com/questions/86148/can-a-3-color-dfs-be-used-to-identify-cycles-not-just-detect-them
+//! [rustc-cycle-detection]: https://github.com/rust-lang/rust/blob/925dc37313853f15dc21e42dc869b024fe488ef3/compiler/rustc_data_structures/src/graph/iterate/mod.rs
+
+use crate::graph::Graph as G;
+use bitvec::vec::BitVec;
+use petgraph::graph::{IndexType, NodeIndex};
+use std::{marker::PhantomData, ops::ControlFlow};
```
Comment (regexident) [inline src/graph/tri_color.rs:21]:
```suggestion
use std::{marker::PhantomData, ops::ControlFlow};

use bitvec::vec::BitVec;
use petgraph::graph::{IndexType, NodeIndex};

use crate::graph::Graph as G;
```

**Review by regexident — COMMENTED** [review]
Hi @smoelius, thank you for taking the time to work on this!

> I included a rustfmt.toml file to make it easier to diff tri_color.rs […] Is that silly? Should we just live with an ugly diff?

I generally prefer to keep `rustfmt`'s config unchanged, but for the sake of diffing with upstream this seems like a reasonable exception to make.

I would however appreciate if you could move `tri_color.rs` and `rustfmt.toml` into a `crate::graph::cycles` module to limit the effect of `rustfmt.toml` to just `crate::graph::cycles::tri_color` and not all of `crate::graph::*`.

The `.github/configs/skywalking-eyes.yml` file will also require an exception for the `tri_color.rs` file:

```toml
  paths:
    - "src/**/*.rs"
  paths-ignore: 👈
    - "src/graph/cycles/tri_color.rs" 👈
```

For the minimal-versions failure you may want to add the following to `Cargo.toml` at line 44:

```toml
# minimal versions
serde_repr = "0.1.12"
wyz = "0.5.1" 👈
```

**Comment by smoelius** [comment]
> I would however appreciate if you could move `tri_color.rs` and `rustfmt.toml` into a `crate::graph::cycles` module to limit the effect of `rustfmt.toml` to just `crate::graph::cycles::tri_color` and not all of `crate::graph::*`.

That absolutely should have occurred to me.

Please tell me what else you think needs to be adjusted.

**Comment by regexident** [comment]
> Please tell me what else you think needs to be adjusted.

The only other thing that comes to my mind right now be adding a comment to `rustfmt.toml` to give some context as to why it's there.

**Comment by smoelius** [comment]
> The only other thing that comes to my mind right now be adding a comment to `rustfmt.toml` to give some context as to why it's there.

Done. No worries if you think of something else.

**Comment by regexident** [comment]
Thanks for working on this @smoelius!

**Pull request merged**